f3/granite.go

```diff
@@ -350,9 +372,99 @@ func (i *instance) isValid(msg *GMessage) bool {
 	return true
 }
 
-// TODO Checks whether a message is justified by prior messages.
-func (i *instance) isJustified(msg *GMessage) bool {
-	return true
+func (i *instance) VerifyJustification(msg *GMessage) error {
+
+	power := NewStoragePower(0)
+	_ = msg.Justification.QuorumSignature.Signers.ForEach(func(bit uint64) error {
+		power.Add(power, i.powerTable.Entries[bit].Power)
```

Comment:
Probably combine this into one iteration, as well as range checks.

Reply:
See above comment.